libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -0,0 +1,264 @@
+    let actual = format(&sem_pkg).unwrap();
+
+    expect.assert_eq(&actual);
+}
+
+#[test]
+fn literals() {
+    let script = r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = -1y2mo3w4d5h6m7s8ms9us10ns
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = -1y2mo3w4d5h6m7s8ms9us10ns:duration
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/"#]],
+    )
+}
+
+#[test]
+fn array_lit() {
+    let script = r#"
+            a = [1, 2, 3]
+            b = [1.1, 2.2, 3.3]
+            c = ["1", "2", "3"]
+            d = [1s, 2m, 3h]
+            e = [2019-10-31T00:00:00Z]
+            f = [/a/, /b/, /c/]
+            g = [{a:0, b:0.0}, {a:1, b:1.1}]
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            a = [1, 2, 3]:[int]
+            b = [1.1, 2.2, 3.3]:[float]
+            c = ["1", "2", "3"]:[string]
+            d = [1s, 2m, 3h]:[duration]
+            e = [2019-10-31T00:00:00Z]:[time]
+            f = [/a/, /b/, /c/]:[regexp]
+            g = [{a: 0, b: 0.0}:{a:int, b:float}, {a: 1, b: 1.1}:{a:int, b:float}]:[{a:int, b:float}]"#]],
+    )
+}
+
+#[test]
+fn dictionary_literals() {
+    let script = r#"
+            a = ["a": 0, "b": 1, "c": 2]
+            b = [1970-01-01T00:00:00Z: 0, 1970-01-01T01:00:00Z: 1]
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            a = ["a": 0, "b": 1, "c": 2]:[string:int]
+            b = [1970-01-01T00:00:00Z: 0, 1970-01-01T01:00:00Z: 1]:[time:int]"#]],
+    )
+}
+
+#[test]
+fn identifier_expressions() {
+    let script = r#"
+            n = 1.0
+            f = n + 3.0
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            n = 1.0
+            f = n:float +:float 3.0"#]],
+    )
+}
+
+#[test]
+fn format_function_expression() {
+    let script = r#"
+            (a) => a
+            f = (a, b=1) => a + b
+            x = f(a:2)
+            y = f(a: x, b: f(a:x))
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            (a) =>{
+            return a:t17
+            }:(a:t17) => t17
+            f = (a, b=1) =>{
+            return a:int +:int b:int
+            }:(a:int, ?b:int) => int
+            x = f:(a:int, ?b:int) => int(a: 2):int
+            y = f:(a:int, ?b:int) => int(a: x:int, b: f:(a:int, ?b:int) => int(a: x:int):int):int"#]],
+    )
+}
+
+#[test]
+fn format_conditional_expression() {
+    let script = r#"
+            if 1 == 2 then 5 else 3
+            ans = if 100 > 0 then "yes" else "no"
+            "#;
+
+    check(
+        script,
+        expect![[r#"
+            package main
+            if 1 ==:bool 2 then 5 else 3
```

Comment:
I think conditional expressions need their own type annotation that shows they type of the value that will be produced when the expression is evaluated. In this case it would be `int`. It might be formatted like this:
```
(if 1 ==:bool 2 then 5 else 3):int
```
I think there needs to be `()` so it's clear that the type annotation refers to the conditional expression. Does that make sense? Can you think of a better way to display the type?